cli/internal/hashing/package_deps_hash.go

```diff
@@ -38,18 +38,29 @@ func GetPackageDeps(rootPath turbopath.AbsolutePath, p *PackageDepsOptions) (map
 		}
 		result = gitLsTreeOutput
 	} else {
+
+		// Add in package.json to input patterns because if the `scripts` in
+		// the package.json change (i.e. the tasks that turbo executes), we want
+		// a cache miss, since any existing cache could be invalid.
+		// Note this package.json will be resolved relative to the pkgPath.
+		p.InputPatterns = append(p.InputPatterns, "package.json")
```

Comment:
This is mutating the `InputPatterns` slice which is explicitly user input. For the inputs into our hash I'd prefer not to mutate the user input (so that we maintain that) and instead create a new slice that has the appended data.

e.g.:
`calculatedInputs := make([]string, len(p.InputPatterns) + 1)`

And then to populate:
```go
for index, glob := range p.InputPatterns {
  calculatedInputs[index] = glob
}
calculatedInputs = append(calculatedInputs, "package.json")
```

We can also use take this opportunity to make sure we don't duplicate user-provided input patterns. `globby` actually has a weird side effect where it will return multiples for matches of different patterns. We run it through a set to dedupe for `outputs` but I'm not actually sure we do it for input patterns so we should take this opportunity to check to make sure that we're properly deduping the glob matches for inputs.

Also worth noting that `append` has one _possibly_ surprising API footgun:
https://go.dev/play/p/_Z-uKzKrNvl

Reply:
Good idea. check out the latest commit and let me know if I've hit any footguns